### ISSUE TITLE
Format Comment RangeError

### ISSUE THREAD

**Issue by tenhobi** [dart-code-checker/dart-code-metrics#1158]
Hi, I am using `dart_code_metrics: ^5.4.0` and this error is throw:

```
fvm dart pub run dart_code_metrics:metrics analyze lib test --fatal-style --fatal-performance --fatal-warnings

⠇ Analyzing 1/2 contexts with 85 file(s)... 5.9sUnhandled exception:
RangeError (index): Invalid value: Valid value range is empty: 0
#0      _StringBase.[] (dart:core-patch/string_patch.dart:260:41)
#1      _Visitor._isValidSentence (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart:114:38)
#2      ListMixin.every (dart:collection/list.dart:138:16)
#3      _Visitor._hasValidMultiline (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart:94:22)
#4      _Visitor.checkRegularComments (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart:56:15)
#5      FormatCommentRule.check (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/format_comment/format_comment_rule.dart:54:8)
#6      LintAnalyzer._checkOnCodeIssues.<anonymous closure> (package:dart_code_metrics/src/analyzers/lint_analyzer/lint_analyzer.dart:278:22)
#7      ExpandIterator.moveNext (dart:_internal/iterable.dart:481:31)
#8      new _GrowableList._ofOther (dart:core-patch/growable_array.dart:202:26)
#9      new _GrowableList.of (dart:core-patch/growable_array.dart:152:26)
#10     new List.of (dart:core-patch/array_patch.dart:51:28)
#11     Iterable.toList (dart:core/iterable.dart:470:12)
#12     LintAnalyzer._checkOnCodeIssues (package:dart_code_metrics/src/analyzers/lint_analyzer/lint_analyzer.dart:283:12)
#13     LintAnalyzer._analyzeFile (package:dart_code_metrics/src/analyzers/lint_analyzer/lint_analyzer.dart:220:21)
#14     LintAnalyzer.runCliAnalysis (package:dart_code_metrics/src/analyzers/lint_analyzer/lint_analyzer.dart:123:26)
<asynchronous suspension>
#15     AnalyzeCommand.runCommand (package:dart_code_metrics/src/cli/commands/analyze_command.dart:53:32)
<asynchronous suspension>
#16     CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#17     CliRunner.run (package:dart_code_metrics/src/cli/cli_runner.dart:56:7)
<asynchronous suspension>
#18     main (file:///Users/tenhobi/.pub-cache/hosted/pub.dartlang.org/dart_code_metrics-5.4.0/bin/metrics.dart:4:3)
<asynchronous suspension>
```

From the stack trace, issue might be here:

https://github.com/dart-code-checker/dart-code-metrics/blob/master/lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart#L54-L56

https://github.com/dart-code-checker/dart-code-metrics/blob/master/lib/src/analyzers/lint_analyzer/rules/rules_list/format_comment/visitor.dart#L114

**Comment by incendial** [comment]
@tenhobi any chance you might know the comment that produces the issue?

**Comment by tenhobi** [comment]
It throws even when we have no format comment warnings left, so I have no clue which comment produces it. :/

**Comment by incendial** [comment]
Fixed in 5.5.0 🚀
